jdaviz/configs/specviz2d/plugins/parsers.py

```diff
@@ -64,7 +65,11 @@ def spec2d_1d_parser(app, data_obj, data_label=None, show_in_viewer=True):
 
         data_obj = Spectrum1D(flux, spectral_axis=spectral_axis, meta=metadata)
 
-    app.data_collection[data_label] = data_obj
+        data_label = return_data_label(app, data_label)
+        app.data_collection[data_label] = data_obj
 
-    if show_in_viewer:
-        app.add_data_to_viewer("spectrum-viewer", data_label)
+        if show_in_viewer:
+            app.add_data_to_viewer("spectrum-viewer", data_label)
+
+    else:
+        raise NotImplementedError("Spectrum2d parser only takes a filename")
```

Comment:
Noticed that Spectrum2d only takes a filename and does not error gracefully if something else is given.